### PR TITLE
Make job config loading respect `.prowignore` files.

### PR DESCRIPTION
**Pull request by cjwagner**
Addresses part 1 of https://github.com/kubernetes/test-infra/issues/24537

/cc @alvaroaleman @chaodaiG

### CODE REVIEW

File: prow/config/config.go [hunk 1]
```diff
@@ -1357,6 +1358,10 @@ func ReadJobConfig(jobConfig string, yamlOpts ...yaml.JSONOpt) (JobConfig, error
 		return jc, nil
 	}
 
+	prowIgnore, err := gitignore.NewRepositoryWithFile(jobConfig, ".prowignore")
```
Comment (petr-muller) [inline prow/config/config.go:1361]: nit: maybe `.prowignore` should be a constant, it's used from two places?

**Review by petr-muller — APPROVED** [review]
/hold

LGTM, holding for assigned reviewers

File: prow/config/config.go [hunk 2]
```diff
@@ -1379,10 +1384,14 @@ func ReadJobConfig(jobConfig string, yamlOpts ...yaml.JSONOpt) (JobConfig, error
 			}
 			return nil
 		}
-
 		if filepath.Ext(path) != ".yaml" && filepath.Ext(path) != ".yml" {
```
Comment (chaodaiG) [inline prow/config/config.go:1387]: understand that this is not from your PR, but I'm pretty sure that `.yml` files are not used by prow, do you know is this expected?

Comment (cjwagner) [reply prow/config/config.go:1387]: Looks like this changed back in 2018 https://github.com/kubernetes/test-infra/pull/9137. I'm pretty sure both are considered valid extensions so its definitely better to respect both.

File: prow/config/config.go (hunk 1 above)
Comment (cjwagner) [reply prow/config/config.go]: Switched to a constant throughout.

**Comment by cjwagner** [comment]
Comments addressed, PTAL.

**Review by petr-muller — APPROVED** [review]

File: prow/config/config.go (hunk 2 above)
Comment (chaodaiG) [reply prow/config/config.go:1389]: ahh.. taking a deeper look and the problem was `config/jobs/**/*.yaml` at https://github.com/kubernetes/test-infra/blob/91011caf8dca5b240f34bd04a4091567cbe44e73/config/prow/plugins.yaml#L516, which is the reason why `.yml` files are not respected

**Review by chaodaiG — APPROVED** [review]

File: prow/config/config.go (hunk 2 above)
Comment (chaodaiG) [reply prow/config/config.go:1389]: btw https://github.com/kubernetes/test-infra/pull/24675

**Comment by cjwagner** [comment]
Thanks!
/hold cancel

**Pull request merged**